data_aug.py

```diff
@@ -0,0 +1,224 @@
+"""
+Title: Audio Data Augmentation and Generator
+Author: Viet Nguyen
+Date: 10/19/2019
+Organization: Cal Poly CSAI
+Description: This library provide 2 classes: AudioAugmenter and DataGenerator.
+"""
+
+import wave
+from pydub import AudioSegment
+from pydub.playback import play
+import numpy as np
+import random
+import math
+import os
+import sys
+
+# Globals for audio specs
+CHUNK_SIZE = 2048
+SAMPLE_WIDTH = 2
+FORMAT = 'int%d' % (SAMPLE_WIDTH * 8)
+NUM_CHANNELS = 1
+SAMPLE_RATE = 16000
+DURATION = 4
+NUMCHUNKS = int(DURATION * SAMPLE_RATE // CHUNK_SIZE)
+BYTES_PER_CHUNK = CHUNK_SIZE * SAMPLE_WIDTH * NUM_CHANNELS
+MAX_PITCH_FACTOR = 200
+MAX_WN_AMP = 20
+MIN_WN_AMP = 0
+MAX_SPEED = 2
+
+# set DURATION to be the actual duration that the model takes
+DURATION = NUMCHUNKS * CHUNK_SIZE / SAMPLE_RATE
+
+
+class AudioAugmenter:
+    """ A wrapper for raw audio that provides methods to augment the audio
+
+        The purpose is to artificially create new data. Every augmentation
+        method returns an AudioAugmenter, which allows for chaining.
+    """
+    def __init__(self, audio=None, raw_bytes=None, sample_width=SAMPLE_WIDTH,
+                 sample_rate=SAMPLE_RATE, num_channels=NUM_CHANNELS):
+        if audio is not None:
+            self.audio = audio
+            self.raw_bytes = audio.raw_data
+
+            self.sample_width = audio.sample_width
+            self.sample_rate = audio.frame_rate
+            self.num_channels = audio.channels
+            self.dtype = 'int%d' % (self.sample_width * 8)
+        elif raw_bytes is not None:
+            self.raw_bytes = raw_bytes
+
+            self.sample_width = sample_width
+            self.sample_rate = sample_rate
+            self.num_channels = num_channels
+            self.dtype = 'int%d' % (self.sample_width * 8)
+
+            self.audio = AudioSegment(data=self.raw_bytes,
+                                      sample_width=self.sample_width,
+                                      frame_rate=self.sample_rate,
+                                      channels=self.num_channels)
+        else:
+            raise ValueError("raw_bytes and audio can not both be None")
+
+    def from_file(path, strict=True):
+        try:
+            audio = AudioSegment.from_file(path)
+        # bad practice but necessary because not obvious what exceptions the
+        # function may raise
+        except:
+            print("Not a valid audio file: %s" % (path))
+            return
```

Comment:
It may be worth it to include the "# nopep8" comment at the end of these lines to suppress the pep8 stylecheck warnings. Source: https://stackoverflow.com/questions/18444840/how-to-disable-a-pep8-error-in-a-specific-file

Reply:
Rest looks good to me

Reply:
I can handle the stylecheck fixes later, approving now